runner/server_workspace_blob.go

```diff
@@ -33,14 +41,48 @@ func (r *TerraformRunnerServer) CreateWorkspaceBlob(ctx context.Context, req *Cr
 		return nil, err
 	}
 
+	secretName := "tf-runner.cache-encryption"
+	encryptionSecretKey := types.NamespacedName{Name: secretName, Namespace: req.Namespace}
+	var encryptionSecret v1.Secret
+
+	log.Info("fetching secret key", "key", encryptionSecretKey)
+	if err := r.Client.Get(ctx, encryptionSecretKey, &encryptionSecret); err != nil {
+		return nil, err
+	}
+
+	// 256 bit AES encryption with Galois Counter Mode.
+	log.Info("encrypting content")
+	token := encryptionSecret.Data["token"]
+	key := token[:EncryptionKeyLength]
+
+	aesCipher, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+
+	gcm, err := cipher.NewGCM(aesCipher)
+	if err != nil {
+		return nil, err
+	}
+
+	nonce := make([]byte, gcm.NonceSize())
+	_, err = rand.Read(nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	out := gcm.Seal(nonce, nonce, blob, nil)
```

Comment:
> // Seal encrypts and authenticates plaintext, authenticates the
	// additional data and appends the result to dst, returning the updated
	// slice. The nonce must be NonceSize() bytes long and unique for all
	// time, for a given key.
	//
	// To reuse plaintext's storage for the encrypted output, use plaintext[:0]
	// as dst. Otherwise, the remaining capacity of dst must not overlap plaintext.
	Seal(dst, nonce, plaintext, additionalData [][byte](https://pkg.go.dev/builtin#byte)) [][byte](https://pkg.go.dev/builtin#byte)

Why do we _need_ to append the result to the `nonce` (1st parameter)? Looking at the example in https://pkg.go.dev/crypto/cipher#example-NewGCM-Encrypt

Reply:
Good question. I don't know :laughing:

Reply:
Oh I know why, because that's how we save it so we can read it back as we don't have any other database or anything to save the nonce. It's there mostly to mess up dictionary attacks. It couldn't be a secret, but not necessarily a secret.

Reply:
When decrypting we can read like this:
```
	nonce, ciphertext := resp.Blob[:nonceSize], resp.Blob[nonceSize:]
```

Without this, we would have to save the nonce somewhere else. Or use a static nonce value which feels a dumb thing to do, we had a secret key for encryption.